cmake/SetupBOUTThirdParty.cmake

```diff
@@ -167,11 +167,19 @@ if (BOUT_USE_NETCDF)
     # Use our own FindnetCDF module which uses nc-config
     find_package(netCDF REQUIRED)
     FetchContent_MakeAvailable(netcdf-cxx4)
+    target_link_libraries(bout++ PUBLIC netCDF::netcdf-cxx4)
   else()
-    find_package(netCDFCxx REQUIRED)
-    set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${netCDF_LIBRARIES}")
+    find_package(netCDFCxx)
+    if (netCDFCxx_FOUND)
+      set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${netCDF_LIBRARIES}")
+      target_link_libraries(bout++ PUBLIC netCDF::netcdf-cxx4)
+    else()
+      find_package(PkgConfig REQUIRED)
+      pkg_check_modules(NETCDF REQUIRED IMPORTED_TARGET netcdf-cxx4)
+      set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${NETCDF_LDFLAGS}")
```

Comment:
Where does `netCDF_CXX_LIBRARY` come from in this case?

Reply:
Reading the docs:
https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_check_modules

It should be `NETCDF_LDFLAGS`.

I would assume `netCDF_CXX_LIBRARY` is not set and doesn't do anything ...